src/toxcore/dht/server.rs

```diff
@@ -123,14 +128,24 @@ impl Server {
             None
         }
     }
+    /// remove timed-out clients
+    pub fn remove_timedout_client(&self) -> IoFuture<()> {
+        let mut state = self.state.write().expect("Failed to obtain ServerState lock");
+        state.peers_cache.retain(|&_pk, ref client|
+            client.last_resp_time.elapsed() <= Duration::from_secs(122));
+        Box::new(future::ok(()))
+    }
     /// send PingRequest to all peer in kbucket    
     pub fn send_pings(&self) -> IoFuture<()> {
-        let mut state = self.state.write().expect("Failed to obtain ServerState lock");
```

Comment:
was it a fix for Lan Discovery? Why do you change this code?

Reply:
For remove timed-out clients, create_client() has been changed.
The change is from (just create client) to (check if it exists in peers_cache).
If client already exists, it means that PingRequest() is sent and client had been created and inserted to peers_cache.
Because ping period is 60 seconds and time-out period is 122 seconds.
If every PingRequest create new client, then new client replace old client with same PK.
To do this, change happen in create_client() and to avoid deadlock send_pings() logic is changed.

Reply:
OK